### PR TITLE
Feature/tags endpoint

### PR DESCRIPTION
**Pull request by yevhen-turovets**
The task in Trello https://trello.com/c/OsnkkjYV

### CODE REVIEW

File: api/app/Repositories/Tag/TagRepository.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace App\Repositories\Tag;
+
+use App\Entity\Tag;
+use App\Repositories\BaseRepository;
+use Illuminate\Support\Collection;
+
+final class TagRepository extends BaseRepository implements TagRepositoryInterface
+{
+    public function get(array $criteria): Collection
```
Comment (dmitriy-uvin) [inline api/app/Repositories/Tag/TagRepository.php:13]: `get` -> `findByCriteria`

File: api/app/Repositories/Tag/TagRepositoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace App\Repositories\Tag;
+
+
+interface TagRepositoryInterface
```
Comment (dmitriy-uvin) [inline api/app/Repositories/Tag/TagRepositoryInterface.php:5]: add `findByCriteria`

File: api/app/Actions/Tag/GetTagsByEventTypeIdAction.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public function execute(GetTagsByEventTypeIdRequest $request): GetTagsByEventTypeIdResponse
+    {
+        $criteria = [];
+
+        if ($request->getEventTypes()) {
+            $criteria[] = new EventTypesCriterion($request->getEventTypes());
+        }
+
+        return new GetTagsByEventTypeIdResponse(
+            $this->repository->get(
```
Comment (dmitriy-uvin) [inline api/app/Actions/Tag/GetTagsByEventTypeIdAction.php:28]: `findByCriteria`

File: api/app/Http/Controllers/Api/TagController.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+namespace App\Http\Controllers\Api;
+
+use App\Actions\Tag\GetTagsByEventTypeIdAction;
+use App\Actions\Tag\GetTagsByEventTypeIdRequest;
+use App\Http\Presenters\TagPresenter;
+use Illuminate\Http\Request;
+
+class TagController extends ApiController
+{
+    private $presenter;
```
Comment (dmitriy-uvin) [inline api/app/Http/Controllers/Api/TagController.php:12]: type for property
TagPresenter $presenter

File: api/routes/api.php [hunk 5]
```diff
@@ -78,6 +78,13 @@
     Route::get('/emails', 'EventController@getEventsEmails');
 });
 
+Route::group([
+                 'namespace' => 'Api\\',
+                 'prefix' => '/tags'
+             ], function () {
+    Route::get('/', 'TagController@getTagsByEventTypeId');
```
Comment (dmitriy-uvin) [inline api/routes/api.php:85]: what method `getTagsByEventTypeId` do?

Comment (aLLeXUs) [reply api/routes/api.php:85]: It depends on how we are gonna use it. If we need we need event type tags - `/event-types/{eventTypeId}/tags`, if we need all tags - `/tags`.

File: api/app/Repositories/Tag/Criterion/EventTypesCriterion.php [hunk 6]
```diff
@@ -10,16 +10,16 @@
 
 final class EventTypesCriterion implements EloquentCriterion
 {
-    private array $eventTypes;
+    private int $eventTypes;
```
Comment (dmitriy-uvin) [inline api/app/Repositories/Tag/Criterion/EventTypesCriterion.php:13]: $eventTypeId

**Review by dmitriy-uvin — APPROVED** [review]

**Pull request merged**